cl/recap/mergers.py

```diff
@@ -950,9 +953,24 @@ async def add_docket_entries(
             await duplicate_rd_queryset.exclude(pk=rd.pk).adelete()
 
         rd.pacer_doc_id = rd.pacer_doc_id or docket_entry["pacer_doc_id"]
-        rd.description = (
-            docket_entry.get("short_description") or rd.description
-        )
+        description = docket_entry.get("short_description") or rd.description
+        if rd.document_type == RECAPDocument.PACER_DOCUMENT:
+            rd.description = description
+        else:
+            rd_qs = de.recap_documents.filter(
+                document_type=RECAPDocument.PACER_DOCUMENT
+            )
+            if await rd_qs.aexists():
+                rd_pd = await rd_qs.afirst()
+                if rd_pd.attachment_number is not None:
+                    continue
+                if rd_pd.description != description:
+                    rd_pd.description = description
+                    try:
+                        await rd_pd.asave()
+                    except ValidationError:
+                        # Happens from race conditions.
+                        continue
```

Comment:
Could you please explain a bit the goal of this block of code?

From what I understand, it attempts to find an attachment that was merged as a `PACER_DOCUMENT` in order to update its description. But why is it selecting the first `PACER_DOCUMENT` in the entry? Which is the same document as in the first `if` statement?

Currently, this code is preventing the descriptions of right attachments from being updated.

Reply:
> But why is it selecting the first PACER_DOCUMENT in the entry?

AFAIU only attachment 0 should be a `PACER_DOCUMENT` `document_type`. But we don't always know which document this is until after attachments has been parsed, so this is supposed to add missing descriptions in cases where the attachment 0 was parsed(say via an attachment page or something) but where the docket doesn't show attachment lists itself.

> But why is it selecting the first `PACER_DOCUMENT` in the entry? Which is the same document as in the first `if` statement?

This is querying against all documents under the docket entry(to essentially check for cases where the docket entry pacer doc id corresponds to attachments other than attachment `0` that were fixed up), while the first `if` statement does not check all documents under the entry but rather only the main entry(which could correspond to any document under the entry).

Reply:
Thanks for the explanation. I see a couple of issues here:

> This is querying against all documents under the docket entry(to essentially check for cases where the docket entry pacer doc id corresponds to attachments other than attachment 0 that were fixed up)

In this case, shouldn't the query:

`rd_qs = de.recap_documents.filter(document_type=RECAPDocument.PACER_DOCUMENT)`

include a `pacer_doc_id` filter that matches the docket entry's `pacer_doc_id` and other conditions within the filter to match the RD that requires to be fixed? This would ensure that the correct `RECAPDocument` to be fixed is selected, and the description is updated correctly for the right document.

Currently, this is selecting the same document as in the previous if statement, which is the main document, and it is incorrectly overwriting its description.

Additionally, the inclusion of this block of code prevents the `RECAPDocument` descriptions for regular attachments (those that don't need fixing) from being properly updated, leaving them blank. Therefore, we would need an else statement to handle those cases.

Reply:
> include a `pacer_doc_id` filter that matches the docket entry's `pacer_doc_id` and other conditions within the filter to match the RD that requires to be fixed? This would ensure that the correct `RECAPDocument` to be fixed is selected, and the description is updated correctly for the right document.

Well..that's where this gets tricky, `pacer_doc_id` will in some cases not give us the correct `RECAPDocument` if the `pacer_doc_id` corresponds with an attachment with a number other than 0. The attachment fixup in theory now should fix things up so that the `rd` with `RECAPDocument.PACER_DOCUMENT` is always the right entry for the short description.

> Additionally, the inclusion of this block of code prevents the RECAPDocument descriptions for regular attachments (those that don't need fixing) from being properly updated, leaving them blank. Therefore, we would need an else statement to handle those cases.

Hmm, regular attachment descriptions I think need to be set in `merge_attachment_page_data`.

Reply:
Got it. The issue I see, which is related to the failing test, comes from the fact that some reports can contain attachments under `docket_entries`, as is the case with the `InternetArchive` report.

Consider the following entries in a report of this type:

```
"docket_entries":[
   {
      "document_number":116,
      "pacer_doc_id":02301124594,
      "attachment_num":0,
      "short_description":"Notice of Case Assignment"
   },
   {
      "document_number":116,
      "pacer_doc_id":02301124595,
      "attachment_num":1,
      "short_description":"Order Referring Case"
   }
]
```

The first entry is merged as a `PACER_DOCUMENT`, and the `short_description` "Notice of Case Assignment" is assigned.

Then when attachment 1 is merged, it enters the condition where the following queryset is executed:
```
 rd_qs = de.recap_documents.filter(
                document_type=RECAPDocument.PACER_DOCUMENT
            )
```

This `queryset` returns the main document `02301124594`, and its `short_description` is incorrectly overwritten with "Order Referring Case", even though this description belongs to attachment 1.

Additionally, the `short_description` for attachment 1 remains blank.

Reply:
Had to fix up a few issues in [#1137](https://github.com/freelawproject/juriscraper/pull/1137) to match the normal parsed docket format but I think it's working now.

Reply:
Great, thanks! Yeah, once that Juriscraper PR is merged, we can do the release and get the latest Juriscraper version into this PR before merging it.

Reply:
Updated to latest juriscraper.